core/request.go

```diff
@@ -166,19 +168,23 @@ func makeRequestProcessor(captureSeq requestSeqCapturer, pendingReq requestlist.
 
 		pendingReq.Add(request)
 
-		if err := applyRequest(request); err != nil {
+		currentView, expectedView, releaseView := viewState.HoldView()
+		defer releaseView()
+
+		if currentView != expectedView {
+			return true, nil
```

Comment:
I think this may be a grey area for that return value. On one hand, the request identifier is captured and the request is added to the pending request set before checking this. On the other hand, the replica has not started timers, nor generated a Prepare message at this point. However, the request identifier was captured as new, and the message was processed as much as the replica could do in its current state...

Reply:
We expect that we will restart to process the request in pending request set after view change completes in some mechanism (not implemented yet), so introducing another branch to return silently might be an option.

Reply:
We could refine diagnostic message logging later, before making the first release. So far, we cannot return `new == false` here because the message is new and not dropped. On the other hand, returning `new == true` causes `Handled ...` message with _debug_ level.